### DISCUSSION TITLE
Does materialize require the dataset to be loaded into memory?

### DISCUSSION THREAD

**Discussion by davidfstein** [pyg-team/pytorch-frame#438] [Q&A]
I have a large dataset, ~30 million x ~500. I would like to stream this data from the disk in batches, but I'm wondering if materialize is compatible with this? Does materialize load the entire data into memory to compute column statistics?

**Comment by akihironitta** [comment]
Currently, yes, unfortunately. To unblock your use case, you'd need to patch `Dataset.materialize` and data loading in a way that you don't have to load the entire dataset in memory when sampling mini-batches.

**Reply by davidfstein** [reply]
Got it thank you. Do you anticipate supporting this use case? I can probably help writing a pull but maybe this is already on the roadmap?

**Reply by akihironitta** [reply]
No, I don't think there's any plan at the moment to support the use case.
